yabt/builders/cpp.py

```diff
@@ -403,3 +405,53 @@ def cpp_lib_builder(build_context, target):
         buildenv_workspace, target.props.cmd_env)
     for obj_file in objects:
         target.artifacts.add(AT.object, obj_file)
+
+
+def calc_hash(json_str):
+    m = md5()
+    m.update(json_str.encode('utf8'))
+    return m.hexdigest()
+
+
```

Comment:
Please *heavily* document the decisions made here: the details are obvious if you understand them and completely cryptic otherwise